src/lib/fetch.ts

```diff
@@ -84,18 +98,44 @@ function getUserAgent() {
   return `rdme${gh}/${pkg.version}`;
 }
 
+/**
+ * Resolves relative path references for local paths,
+ * otherwise returns the path
+ */
+function normalizeFilePath(opts: FilePathDetails) {
+  if (opts.fileType === 'path') return path.relative('', opts.filePath);
```

Comment:
Why do we need to do this? Say a user passes in a file called `./documentation/rdme.md`. Instead of generating something like this:

```
https://github.com/readmeio/rdme/blob/cb4129d5c7b51ff3b50f933a9c7d0c3d0d33d62c/./documentation/rdme.md
```

we resolve the relative paths and produce this instead:

```
https://github.com/readmeio/rdme/blob/cb4129d5c7b51ff3b50f933a9c7d0c3d0d33d62c/documentation/rdme.md
```

Reply:
This `path.relative()` work will have a lot of quirks if `filePath` doesn't look exactly like `./documentation/rdme.md`:

```js
console.log(path.relative('', './documentation/rdme.md'));
console.log(path.relative('', '/documentation/rdme.md'));
console.log(path.relative('', 'documentation/rdme.md'));
console.log(path.relative('', '../documentation/rdme.md'));
```

```
documentation/rdme.md
../../../../../documentation/rdme.md
documentation/rdme.md
../documentation/rdme.md
```

You might want to do something like this:

```js
function sanitizeRelativePath(filePath) {
  return filePath
    .split('/')
    .filter(Boolean)
    .filter(c => c !== '.' && c !== '..')
    .join('/');
}

console.log(path.relative('', sanitizeRelativePath('./documentation/rdme.md')));
console.log(path.relative('', sanitizeRelativePath('/documentation/rdme.md')));
console.log(path.relative('', sanitizeRelativePath('documentation/rdme.md')));
console.log(path.relative('', sanitizeRelativePath('../documentation/rdme.md')));
```

```
documentation/rdme.md
documentation/rdme.md
documentation/rdme.md
documentation/rdme.md
```

Reply:
So considering that this normalization only takes place when someone is syncing within a GitHub Actions environment, are they ever not running `rdme` from the root of their repo? This could be famous last words™️ but I can't imagine a scenario in which someone is doing something like this:

```yml
- uses: readmeio/rdme@v8
  with:
    rdme: docs ../documentation/rdme.md
```

Is it even possible to reference a file like that unless you're running some arcane subdirectory workflows? If `rdme` can't find the file, it wouldn't even make the fetch request. But if it can locate the file, we wouldn't want to strip the trailing `../` right? In that case, we might want to do some path resolution from the root of the git repo, but I feel like that's a bridge we should cross if customers ask for it.

Reply:
Extreme edge case but what if you run the command with a working directory and then reference a file outside of that?

Reply:
Probably but I don't know if people do that in a GitHub Actions runner right?

Reply:
What about the `--working-directory` arg on rdme?

I think what you've got is pretty solid fwiw as this work is just for headers I don't think it's the end of the world if we don't catch every weird way people might supply a file or directory.

Reply:
Ohhh forgot about that flag. Oof. In that case we might want to do some path resolution from the root of the git repo 😕 I'll think on this more and get back to you!

Reply:
okay I feel good about this approach: https://github.com/readmeio/rdme/pull/735/commits/3093bcadd5f311bf0e812edf9cc59da124ffb583

let me know what y'all think!